jwst/datamodels/model_base.py

```diff
@@ -102,9 +102,10 @@ def __init__(self, init=None, schema=None, extensions=None,
         # Load the schema files
         if schema is None:
             schema_path = os.path.join(base_url, self.schema_url)
-            extension_list = asdf_extension.AsdfExtensionList(self._extensions)
+            # Create an AsdfFile so we can use its resolver for loading schemas
+            asdf_file = AsdfFile(extensions=self._extensions)
             schema = asdf_schema.load_schema(schema_path,
-                resolver=extension_list.url_mapping, resolve_references=True)
+                resolver=asdf_file.resolver, resolve_references=True)
 
```

Comment:
Apparently `extension_list.url_mapping` returns a resolver as well.

Reply:
It does, but the extension list that was used previously did not include ASDF's built-in extension. Doing it this way ensures that the built-in extension is included automatically. Maybe ASDF could provide a better API for this in the future, though.

Reply:
Also, I believe that the previous way would not have properly handled tag references in schemas, whereas this new way will.